server/src/test/java/io/crate/auth/HostBasedAuthenticationTest.java

```diff
@@ -545,12 +562,122 @@ public void cert_method_resolved_when_ssl_on_and_keystore_configured() throws Ex
             SystemDefaultDnsResolver.INSTANCE,
             () -> "dummy"
         );
+        SSLSession sslSession = mock(SSLSession.class);
+        when(sslSession.getPeerCertificates()).thenReturn(new Certificate[]{mock(Certificate.class)});
+
         AuthenticationMethod authMethod = hba.resolveAuthenticationType("crate",
-            new ConnectionProperties(InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, mock(SSLSession.class)));
+            new ConnectionProperties(CRATE_USER_CREDENTIALS, InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, sslSession));
         assertThat(authMethod).isExactlyInstanceOf(ClientCertAuth.class);
 
         AuthenticationMethod authMethod2 = hba.resolveAuthenticationType("crate",
-            new ConnectionProperties(InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, mock(SSLSession.class)));
+            new ConnectionProperties(CRATE_USER_CREDENTIALS, InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, sslSession));
         assertThat(authMethod2).isExactlyInstanceOf(ClientCertAuth.class);
     }
+
+    @Test
+    public void test_multiple_matches_method_uniquely_identified_by_properties() {
+        // Validating different declaration order as auth used to fail depending on order.
+        Settings pwdFirst = Settings.builder()
+            .put("auth.host_based.config.1.method", "password")
+            .put("auth.host_based.config.2.method", "jwt")
+            .put("auth.host_based.config.2.protocol", "http")
```

Comment:
Could be removed as not related.
```suggestion
```

Reply:
we must have http with jwt, otherwise it won't pass validation.

https://github.com/crate/crate/commit/e8b7e320e3ba7d1d5d1fb8ce4ebb7c4c151784ad#diff-8969849af4e14121fe01f114f66b4e49f995a503d2efaade24affd7699c42415R147-R148